### DISCUSSION TITLE
API Rules for multiple auth collections

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#2852] [Q&A] [answered]
@ganigeorgiev I just realized that there is a level of implicitness in `@request.auth.id` that could be a problem if there is more than one auth collection.  How would that work out?  Maybe it should be `@auth.user.id` so that other collections could be referenced?  It logically goes under `@request` due to statelessness, but maybe there could be a shortcut?

**Comment by ganigeorgiev** [comment, marked as answer]
This is not documented very well, but on application level we have a validator that makes sure each auth id is unique across all auth collections (aka. you shouldn't be able to save for example `managers` and `clients` with the same id).

To differentiate the current authenticated auth collection in an API rule you can use the `@request.auth.collectionId` or `@request.auth.collectionName` fields.
